src/main/java/org/broadinstitute/hellbender/tools/walkers/varianteval/util/VariantEvalUtils.java

```diff
@@ -310,4 +310,13 @@ private void addMapping(HashMap<String, Collection<VariantContext>> mappings, St
             mappings.put(sample, new ArrayList<>(1));
         mappings.get(sample).add(vc);
     }
+
+    /**
+     * Returns a map of all detected VariantStratifier classes
+     *
+     * @return An unmodifiable map of all VariantStratifier classes
+     */
+    public static Map<String, Class<? extends VariantStratifier>> getStratifierClasses() {
```

Comment:
Since this isn't required, lets remove it. To answer your question, I would generally avoid handing out internal structures like this, especially Maps, since its not clear from the types what it contains. There are several alternative ways to do this, and I know this code already does similar things all over the place (extreme example being bindVariantContexts), but lets not expose any more if we can avoid it. As you mentioned, the refactoring is the right way to address it - I think that would be a fair amount of work, but if we want to continue to make changes to this code it will probably be necessary.

Reply:
ok